src/main/java/edu/harvard/iq/dataverse/DOIDataCiteServiceBean.java

```diff
@@ -171,33 +178,79 @@ public void deleteRecordFromCache(Dataset datasetIn){
+        try {
+            doiMetadata = doiDataCiteRegisterService.getMetadata(identifier);
+        } catch (Exception e) {
+            logger.log(Level.WARNING, "deleteIdentifier: get matadata failed. " + e.getMessage(), e);
+        }
+
+        String idStatus = doiMetadata.get("_status");
+        if ( idStatus != null ) {
+            switch ( idStatus ) {
+                case RESERVED:
+                case DRAFT:    
+                    logger.log(Level.INFO, "Delete status is reserved..");
+                    try {
+                    	//service only removes the identifier from the cache (since it was written before DOIs could be registered in draft state)
+                        doiDataCiteRegisterService.deleteIdentifier(identifier);
+                        //So we call the deleteDraftIdentifier method below until things are refactored
+                        deleteDraftIdentifier(dvObject);
+                    } catch (Exception e) {
+                        logger.log(Level.WARNING, "delete failed: " + e.getMessage(), e);
+                    }
+                    break;
+
+                case PUBLIC:
+                case FINDABLE:
+                    //if public then it has been released set to unavailable and reset target to n2t url
+                    Map<String, String> metadata = addDOIMetadataForDestroyedDataset(dvObject);
+                    metadata.put("_status", "registered");
+                    metadata.put("_target", getTargetUrl(dvObject));                   
+                    doiDataCiteRegisterService.deactivateIdentifier(identifier, metadata, dvObject);
+                    break;
+            }
+        }
+    }
+        
     /**
      * Deletes DOI from the DataCite side, if possible. Only "draft" DOIs can be
      * deleted.
      */
-    @Override
-    public void deleteIdentifier(DvObject dvObject) throws IOException {
+    private void deleteDraftIdentifier(DvObject dvObject) throws IOException {
+    	
+    	//ToDo - incorporate into DataCiteRESTfulClient
         String baseUrl = System.getProperty("doi.baseurlstringnext");
         String username = System.getProperty("doi.username");
         String password = System.getProperty("doi.password");
-        String pid = dvObject.getGlobalId().asString();
-        int result = PidUtil.deleteDoi(pid, baseUrl, username, password);
-        logger.fine("Result of deleteIdentifier for " + pid + ": " + result);
-    }
+        GlobalId doi = dvObject.getGlobalId();
+        /**
+         * Deletes the DOI from DataCite if it can. Returns 204 if PID was deleted
+         * (only possible for "draft" DOIs), 405 (method not allowed) if the DOI
+         * wasn't deleted (because it's in "findable" state, for example, 404 if the
+         * DOI wasn't found, and possibly other status codes such as 500 if DataCite
+         * is down.
+         */
 
-    private boolean updateIdentifierStatus(DvObject dvObject, String statusIn) {
```

Comment:
@qqmyers I don't think we talked about this `updateIdentifierStatus` method. Was it simply unused?

Reply:
Yeah - looks like this private method exists in the EZID and Handle code, called by the public publicizeIdentifier() method but the DataCite code just have the same code directly in the publicizeIdentifier method.